lang/c++/test/SchemaTests.cc

```diff
@@ -199,7 +199,32 @@ const char* roundTripSchemas[] = {
     "{\"type\":\"fixed\",\"name\":\"Test\",\"size\":1}"
 };
 
+const char* schemasToCompact[] = {
+    // Schema without any whitespace
+    "{\"type\":\"record\",\"name\":\"Test\",\"fields\":[]}",
+
+    // Schema with whitespaces outside of field names/values only.
+    "{\"type\":   \"record\",\n   \n\"name\":\"Test\", \t\t\"fields\":[]}\n \n",
 
+    // Schema with whitespaces both inside and outside of field names/values.
+    "{\"type\":   \"record\",  \"name\":               \"ComplexInteger\"\n, "
+    "\"doc\": \"record_doc °C \u00f8 \u001f \\n \n \t\", "
```

Comment:
My compiler on Mac does not like unicode escapes for single byte characters. Replacing `\u001f` with `\x1f` fixes it for me.